utils/assign_components.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/python3
+
+import argparse
+import collections
+import csv
+import re
+import subprocess
+import yaml
+
+from pathlib import Path
+
+import ssg.build_yaml
+import ssg.environment
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--build-config-yaml")
+    parser.add_argument("--product-yaml")
+    parser.add_argument("--target-dir")
+    parser.add_argument("--benchmark-root")
+    args = parser.parse_args()
+    return args
+
+
+def get_all_groups(benchmark_root):
+    groups = []
+    for group_yml in benchmark_root.glob("**/group.yml"):
+        groups.append(group_yml.parent.name)
+    return sorted(groups)
+
+
+def get_group_to_components_mapping():
+    group_to_components = {}
+    with open("utils/group_to_component.csv") as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            group = row[0]
+            components = set(row[1:])
+            group_to_components[group] = components
+    return group_to_components
+
+
+def get_package_to_components_mapping():
+    package_to_component = {}
+    with open("utils/component_to_package.csv") as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            component = row[0]
+            packages = set(row)
+            for package in packages:
+                package_to_component[package] = component
+    return package_to_component
+
+
+def get_component_to_package_mapping():
+    component_to_package = {}
+    with open("utils/component_to_package.csv") as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            component = row[0]
+            packages = list(row)
+            component_to_package[component] = packages
+    return component_to_package
+
+
+def get_components_by_groups(rule_groups, group_to_components):
+    components = set()
+    for group in rule_groups:
+        components |= group_to_components.get(group, set())
+    return components
+
+
+template_to_component = {
```

Comment:
WDYT of moving this directly to templates (in the future)?

Reply:
I plan to do that